doniclient/osc/cli.py

```diff
@@ -115,246 +74,226 @@ def take_action(self, parsed_args):
-            ),
-            required=True,
+        properties = parser.add_argument_group("properties")
+
+        properties.add_argument(
+            "--mgmt_addr",
+            metavar="<mgmt_addr>",
+            dest="properties.mgmt_addr",
+            action=GroupedAction,
+        )
+        properties.add_argument(
+            "--ipmi_username",
+            metavar="<ipmi_username>",
+            dest="properties.ipmi_username",
+            action=GroupedAction,
+        )
+        properties.add_argument(
+            "--ipmi_password",
+            metavar="<ipmi_password>",
+            dest="properties.ipmi_password",
+            action=GroupedAction,
+        )
+        properties.add_argument(
+            "--ipmi_terminal_port",
+            metavar="<ipmi_terminal_port>",
+            dest="properties.ipmi_terminal_port",
+            action=GroupedAction,
+            type=int,
+        )
+        properties.add_argument(
+            "--deploy_kernel",
+            metavar="<deploy_kernel>",
+            dest="properties.baremetal_deploy_kernel_image",
+            action=GroupedAction,
+        )
+        properties.add_argument(
+            "--deploy_ramdisk",
+            metavar="<deploy_ramdisk>",
+            dest="properties.baremetal_deploy_ramdisk_image",
+            action=GroupedAction,
+        )
+        properties.add_argument(
+            "--ironic_driver",
+            metavar="<ironic_driver>",
+            dest="properties.baremetal_driver",
+            action=GroupedAction,
         )
+        properties.add_argument(
+            "--resource_class",
+            metavar="<resource_class>",
+            default="baremetal",
+            dest="properties.baremetal_resource_class",
+            action=GroupedAction,
+        )
+        properties.add_argument(
+            "--cpu_arch",
+            metavar="<cpu_arch>",
+            default="x86_64",
+            dest="properties.cpu_arch",
+            action=GroupedAction,
+        )
+        properties.add_argument(
+            "--blazar_node_type",
+            metavar="<blazar_node_type>",
+            dest="properties.node_type",
+            action=GroupedAction,
+        )
+        properties.add_argument(
+            "--blazar_su_factor",
+            metavar="<blazar_su_factor>",
+            dest="properties.su_factor",
+            type=float,
+            default=1.0,
+            action=GroupedAction,
+        )
+        properties.add_argument(
+            "--placement",
+            metavar="<placement>",
+            dest="properties.placement",
+            action=GroupedAction,
+            type=json.loads,
+        )
+        properties.add_argument(
+            "--capabilities",
+            metavar="<capabilities>",
+            dest="properties.baremetal_capabilities",
+            action=GroupedAction,
+            type=json.loads,
+        )
+
+        properties.add_argument(
+            "--interfaces",
```

Comment:
I'm beginning to be of the opinion that we should drop all of these flags and just have `--property` or `--properties`. It feels a bit unnecessary to have to add them to the client when we add support to the server. But, I guess explicit flags aid in discovery.

Reply:
It's helpful mostly because the same parser is used for update, and it's common to want to update a single field, or add an unset property.